### PR TITLE
Handle wildcards in vs destinationhost analyzer

### PR DESCRIPTION
**Pull request by sushicw**
Handle wildcards correctly for destinationhost analyzer. Fixes https://github.com/istio/istio/issues/17976

### CODE REVIEW

**Comment by sushicw** [comment]
> 🤔 🐛 You appear to be fixing a bug in Go code, yet your PR doesn't include updates to any test files. Did you forget to add a test?
> 

The change includes a new test case.

File: galley/pkg/config/analysis/analyzers/testdata/virtualservice_destinationhosts.yaml [hunk 1]
```diff
@@ -14,6 +14,15 @@ spec:
   - external-reviews.org
 ---
 apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: eu-wildcard
+  namespace: default
+spec:
+  hosts:
+  - "*.eu.bookinfo.com"
```
Comment (nrjpoddar) [inline galley/pkg/config/analysis/analyzers/testdata/virtualservice_destinationhosts.yaml]: Can we add another test case for hosts being equal to `"*.bookinfo.com"`

**Comment by nrjpoddar** [comment]
> /retest

@sushicw some of my PRs also end up ping-ponging between tests randomly failing which is quite frustrating as it slows you down significantly. Do you know if anyone is working on addressing/disabling flakes?

**Comment by sushicw** [comment]
@nrjpoddar Yes, there's definitely a lot of attention going to fixing flaky tests (and more often, flakiness in the test infra).

**Comment by sushicw** [comment]
Rebased against https://github.com/istio/istio/pull/17977. This PR is now just some incremental structural cleanup, keeping the bits of my PR that I think are worth keeping on top of @burmanm 's work.

**Pull request merged**